zebROS_ws/src/behaviors/src/path_to_apriltag.cpp

```diff
@@ -0,0 +1,251 @@
+#include "ros/ros.h"
+#include "behavior_actions/GamePiecePickup.h"
+#include <nav_msgs/Path.h>
+#include <actionlib/client/simple_action_client.h>
+#include <actionlib/client/terminal_state.h>
+#include <path_follower_msgs/PathAction.h>
+#include "base_trajectory_msgs/GenerateSpline.h"
+#include <field_obj/Detection.h>
+#include "tf2_ros/transform_listener.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.h"
+#include "geometry_msgs/PoseWithCovarianceStamped.h"
+#include <optional>
+#include "behavior_actions/PathToAprilTagAction.h"
+#include <sensor_msgs/Imu.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2/LinearMath/Vector3.h>
+#include <actionlib/server/simple_action_server.h>
+
+/*
+"object_id: '1'
+max_objects: 1
+primary_frame_id: 'base_link'
+secondary_object_id: ''
+secondary_max_objects: 0
+secondary_max_distance: 0.0
+secondary_frame_id: ''
+min_radius: 0.0
+endpoint:
+    position: {x: 0.0, y: 0.0, z: 0.0}
+    orientation: {x: 0.0, y: 0.0, z: 0.0, w: 0.0}"
+*/
+
+trajectory_msgs::JointTrajectoryPoint generateTrajectoryPoint(double x, double y, double rotation)
+{
+	trajectory_msgs::JointTrajectoryPoint point;
+	point.positions.resize(3);
+	point.positions[2] = rotation;
+	point.positions[0] = x;
+	point.positions[1] = y;
+	return point;
+}
+
+base_trajectory_msgs::GenerateSpline makePath(double x, double y, double rotation, std::string frame_id) {
+    base_trajectory_msgs::GenerateSpline srv;
+    srv.request.header.stamp = ros::Time::now();
+    srv.request.header.frame_id = "base_link";
```

Comment:
Wouldn't we want to define a frame_id for the intake and drive it there rather than the center of the robot?

Reply:
yeah that's probably a better idea, where should we define the frame? maybe another node to publish the transform based on whether the intake is in or out?

(also I just realized that function is never used so am deleting it, the actual path generation is in `executeCB`)

Reply:
We could just define 3 static transforms for the 3 levels of arm placing. We wouldn't necessarily want to track to the current state since once we drive to place the arm would move anyway?

Reply:
wait we don't need a separate transform for low medium and high right? since we ignore z?

Reply:
or do you mean that it has three horizontal positions?

Reply:
Thinking about it, we might not need any of them
The robot's going to be in front of each position, so that's the x,y coordinate we need to go to regardless of which height we're placing at.  Those coordinates could just be hard-coded to an x,y coordinate the center of the drive base needs to be at so the front bumper is as far towards the driver's station as possible. From there, the placing behavior would be responsible for getting the elevator to the correct position?

I mean, maybe we hard-code x,y offsets, maybe we use a "base_link->center of front bumper" transform to do the math?